src/servers/src/http/loki.rs

```diff
@@ -224,9 +237,21 @@ async fn handle_json_req(
                 stream_index,
                 line_index
             );
-            // TODO(shuiyisong): we'll ignore structured metadata for now
 
-            let mut row = init_row(schemas.len(), ts, line_text);
+            let structured_metadata = line
+                .get(2)
+                .and_then(|sdata| sdata.as_object().cloned())
+                .unwrap_or_default()
+                .into_iter()
+                .filter_map(|(k, v)| {
+                    v.as_str()
+                        .map(|v| (k.clone(), Value::String(v.to_string().into())))
+                })
+                .collect::<BTreeMap<String, Value>>();
```

Comment:
it seems we cloned all data twice with `sdata.as_object().cloned()` and then k/v clone/tostring. Is it possible to remove first clone on `as_object()`